Venmo/src/main/java/com/braintreepayments/api/VenmoClient.java

```diff
@@ -428,6 +447,219 @@ private Intent getLaunchIntent(Configuration configuration, String profileId, St
+        onBrowserSwitchResult(browserSwitchResult, new VenmoOnActivityResultCallback() {
+            @Override
+            public void onResult(@Nullable VenmoAccountNonce venmoAccountNonce, @Nullable Exception error) {
+                if (listener != null) {
+                    if (venmoAccountNonce != null) {
+                        listener.onVenmoSuccess(venmoAccountNonce);
+                    } else if (error != null) {
+                        listener.onVenmoFailure(error);
+                    }
+                }
+            }
+        });
+
+        this.pendingBrowserSwitchResult = null;
+    }
+
+    @VisibleForTesting
+    void startAppLinkFlow(FragmentActivity activity, VenmoIntentData input) throws JSONException, BrowserSwitchException {
+        JSONObject braintreeData = new MetadataBuilder()
+                .sessionId(input.getSessionId())
+                .integration(input.getIntegrationType())
+                .version()
+                .build();
+
+        String applicationName = "ApplicationNameUnknown";
+        Context context = braintreeClient.getApplicationContext();
+        if (context != null) {
+            if (context.getPackageManager().getApplicationLabel(context.getApplicationInfo()).toString() != null) {
+                applicationName = context.getPackageManager().getApplicationLabel(context.getApplicationInfo()).toString();
+            }
+        }
+
+        Uri venmoBaseURL = Uri.parse("https://venmo.com/go/checkout")
+                .buildUpon()
+                .appendQueryParameter("x-success", braintreeClient.getReturnUrlScheme() + "://x-callback-url/vzero/auth/venmo/success")
+                .appendQueryParameter("x-error", braintreeClient.getReturnUrlScheme() + "://x-callback-url/vzero/auth/venmo/error")
+                .appendQueryParameter("x-cancel", braintreeClient.getReturnUrlScheme() + "://x-callback-url/vzero/auth/venmo/cancel")
+                .appendQueryParameter("x-source", applicationName)
+                .appendQueryParameter("braintree_merchant_id", input.getProfileId())
+                .appendQueryParameter("braintree_access_token", input.getConfiguration().getVenmoAccessToken())
+                .appendQueryParameter("braintree_environment", input.getConfiguration().getVenmoEnvironment())
+                .appendQueryParameter("resource_id", input.getPaymentContextId())
+                .appendQueryParameter("braintree_sdk_data", braintreeData.toString())
+                .appendQueryParameter("customerClient", "MOBILE_APP")
+                .build();
+
+        BrowserSwitchOptions browserSwitchOptions = new BrowserSwitchOptions()
+                .requestCode(BraintreeRequestCodes.VENMO)
+                .url(venmoBaseURL)
+                .returnUrlScheme(braintreeClient.getReturnUrlScheme());
+
+        braintreeClient.startBrowserSwitch(activity, browserSwitchOptions);
+        braintreeClient.sendAnalyticsEvent("pay-with-venmo.browser-switch.started");
```

Comment:
I think we'll want a way to differentiate the old vs new switch methods in our analytics. Maybe that's in an `additionalDetails` tag? (It could also cover the PP checkout vs vault distinction we also wanted to cover at some point too)

Reply:
I went with the "browser-switch" name because that's what all of the other modules use, but they only support 1 browser switch method vs app links that this module supports for both flows. But we could update this to "app-links" instead. This method is used for both the app and web fallback flow via app links.

Reply:
Ah - I'm remembering that this version is still on Arachne. So adding a new tag name in the POST body means an Arachne deploy ...

So adding a new event name to distinguish the 2 switch types is probably our best bet

Reply:
Yeah, still on arachne, but it is a good point since we will need to port this to v5. The intent based flow (old flow) uses: `"pay-with-venmo.app-switch.<ACTION>"` as the event name so that naming is totally seperate. 

Do you think we should go with `"pay-with-venmo.browser-switch.<ACTION>"` to remain consistent across v4 modules or update this PR to use `"pay-with-venmo.app-link.<ACTION>"` to prepare for what we may want to call this in v5?

Reply:
Great question! Hm - I lean toward moving it closer to the v5 naming convention. But really as long as we document it, either should be usable.

Maybe we can add a new tabbed section to our events Excel doc to call this out (since the existing tab is for V5+)

Reply:
Done: https://github.com/braintree/braintree_android/pull/911/commits/85114837ccefb1e73efef3ffeb02c709cb2e3d7a

Also added a new tab for Android v4 on the excel. Though these are the only events there for now, but at least we will have somewhere to point to for the event names if needed! Not sure if we plan to add iOS v5 + the rest of the Android v4 events at some point. Could be a good "tech debt" task at some point so we can point folks with questions on those versions somewhere.